crates/proto/src/rr/dnssec/rdata/rrsig.rs

```diff
@@ -69,6 +69,34 @@ impl RRSIG {
             sig,
         ))
     }
+
+    /// Returns the authenticated TTL of this RRSIG with a Record.
+    ///
+    /// ```text
+    /// RFC 4035             DNSSEC Protocol Modifications            March 2005
+    ///
+    /// If the resolver accepts the RRset as authentic, the validator MUST
+    /// set the TTL of the RRSIG RR and each RR in the authenticated RRset to
+    /// a value no greater than the minimum of:
+    ///
+    ///   o  the RRset's TTL as received in the response;
+    ///
+    ///   o  the RRSIG RR's TTL as received in the response;
+    ///
+    ///   o  the value in the RRSIG RR's Original TTL field; and
+    ///
+    ///   o  the difference of the RRSIG RR's Signature Expiration time and the
+    ///      current time.
+    /// ```
+    ///
+    /// See RFC 4035, section 5.3.3: https://datatracker.ietf.org/doc/html/rfc4035#section-5.3.3
+    ///
+    pub fn authenticated_ttl(&self, record: &crate::rr::Record, current_time: u32) -> u32 {
```

Comment:
Nit: any particular reason not to import `Record` here? Would be nicer IMO.

Reply:
Updated.